api/proto/canary/newrelic.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+package proto.canary;
+
+option go_package = "github.com/spinnaker/kleat/api/client/canary";
+
+// Configuration for the New Relic canary integration.
+message NewRelic {
+  // Whether the integration is enabled.
+  bool enabled = 1;
+
+  // The list of configured accounts.
+  repeated NewRelicAccount accounts = 2;
+}
+
+message NewRelicAccount {
+  // The name of the account.
+  string name = 1;
+
+  // Configuration for the New Relic Insights server endpoint.
+  Endpoint endpoint = 2;
+
+  // (Required) Your account's unique New Relic Insights API key. See
+  // https://docs.newrelic.com/docs/insights/insights-api/get-data/query-insights-event-data-api.
+  string apiKey = 3;
+
+  // (Required) Your New Relic account ID. See
+  // https://docs.newrelic.com/docs/accounts/install-new-relic/account-setup/account-id.
+  string applicationKey = 4;
+
+  // Configuration for the New Relic Insights server endpoint.
+  message Endpoint {
```

Comment:
I wonder if this could become a top-level message as this is the second account I've seen it in to date.

Reply:
I thought about that too. I think when we consider sharing messages we need to consider the most common cases of if/how these configuration parameters will be updated. I think sharing an Endpoint message would make sense if we anticipate a future where someone refactors all the client code at once that uses the current structure. That would be a nice future. But I think the future we'll get is one-off changes from provider maintainers who will only update the calling code for their particular integration, and won't even have a way to test other integrations.

Reply:
That's a fair point, I guess for now it's only on field under this but perhaps in the future there might be integration-specific fields.